tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -365,6 +366,19 @@ void handleRequestImpl(sourcekitd_object_t ReqObj, ResponseReceiver Rec) {
     return Rec(RB.createResponse());
   }
 
+  if (ReqUID == RequestCompilerVersion) {
+    ResponseBuilder RB;
+    auto dict = RB.getDictionary();
+    auto thisVersion = swift::version::Version::getCurrentLanguageVersion();
+    dict.set(KeyVersionMajor, static_cast<int64_t>(thisVersion[0]));
+    dict.set(KeyVersionMinor, static_cast<int64_t>(thisVersion[1]));
+    if (thisVersion.size() > 2)
+      dict.set(KeyVersionPatch, static_cast<int64_t>(thisVersion[2]));
+    else
+      dict.set(KeyVersionPatch, static_cast<int64_t>(0));
```

Comment:
not sure if we should do it or just skip the key - I've preferred to default to 0 so the test could test the presence of `key.version_patch`

Reply:
Can we drop `key.version_patch` altogether? MajorVersion & minorVersion seems to suffice for me. See here: https://github.com/apple/swift/blob/master/include/swift/Basic/Version.h#L174

Reply:
I'd prefer to expose more information than needed than finding ourselves on a situation where we need that information and it's missing. 

Historically, SwiftLint has never needed to check the patch version, but I can imagine cases where we could need it (e.g. a feature is introduced on a minor version, but has a bug that prevents us to rely on it and it gets fixed on a patch later).

Reply:
The patch version is useful in cases where we'd want to make this compiler version request. Say we have a workaround for a bug in Swift 5.0.0 that was fixed in 5.0.1 (something like #22090). It would be useful to fork based on this value.

Reply:
🤔Thinking more about this, If we'd like to check bug fixes, isn't it better to check SourceKit dylib's version number? do we really need language versions for this purpose? 
SourceKit has a version number defined here: https://github.com/apple/swift/blob/master/tools/SourceKit/CMakeLists.txt#L7

Reply:
That would work just as well I think.

Reply:
> SourceKit has a version number defined here: https://github.com/apple/swift/blob/master/tools/SourceKit/CMakeLists.txt#L7

That information can not be used on Linux version.
Even if we use `CFBundleVersion` in `sourcekitd.framework` on macOS, it seems very difficult to apply workarounds for bugs based on that information.
```terminal.sh-session
$ (find /Library/Developer -name sourcekitd.framework; find /Applications -name sourcekitd.framework) 2>/dev/null|xargs -n 1 -t -I % /usr/libexec/PlistBuddy -c "print :CFBundleVersion" %/Versions/A/Resources/Info.plist
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Library/Developer/Toolchains/swift-4.1.3-RELEASE.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
4.1.32018072701
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Library/Developer/Toolchains/swift-4.2.1-RELEASE.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
4.212018103001
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Library/Developer/Toolchains/swift-4.2.2-RELEASE.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
4.222019020401
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Library/Developer/Toolchains/swift-5.0-DEVELOPMENT-SNAPSHOT-2019-02-03-a.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
5.02019020301
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Library/Developer/Toolchains/swift-4.0.3-RELEASE.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
4.032017120501
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
1000.11.42
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Applications/Xcode_9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
902.0.54
/usr/libexec/PlistBuddy -c print :CFBundleVersion /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/Resources/Info.plist
1001.0.45.7
```

Reply:
Using the compiler version is also more understandable from a user perspective: we can say "this rule is only available if you're compiling with Swift 5.0 or above".